### DISCUSSION TITLE
Auto command doesn't seem to run

### DISCUSSION THREAD

**Discussion by dmyyy** [LazyVim/LazyVim#4454] [Q&A]
Trying to run the `ObsidianToday` command on startup when in a specific directory. I'm working off of a base lazy nvim starter config. The auto command seems to be running but not doing anything.

Code

```
    OBSIDIAN_WORKSPACES = {
      ["path/obsidian-git-sync"] = true,
    }

    -- Run ObsidianToday command if in obsidian workspace
    vim.api.nvim_create_autocmd("VimEnter", {
      callback = function()
        local current_dir = vim.fn.getcwd()
       
        if OBSIDIAN_WORKSPACES[current_dir] then
          vim.cmd("ObsidianToday")
        end
      end,
    })
```

When running nvim with `nvim -V1` and checking the `au VimEnter` command it looks like it is running. Looking through lazy I can't see any print messages or logs. 
```
    VimEnter
    *<Lua 422: ~/.config/nvim/lua/config/autocmds.lua:13>
	    Last set from ~/.config/nvim/lua/config/autocmds.lua line 12
```

Debugging tips or advice would are appreciated. Is there a way to see logs/print statements from auto commands?

The plugin is loaded at start time - maybe auto command is being triggered before plugin is fully loaded?

**Comment by dpetka2001** [comment]
For autocmds on `VimEnter` try to put them in `options.lua` instead of `autocmds.lua`.

**Reply by dmyyy** [reply]
Tried `options.lua` - same result as before. Can verify it seems to run, but nothing happens. Anyway I can see print statements inside the callback?

```

--- Autocommands ---
VimEnter
*<Lua 38: ~/.config/nvim/lua/config/options.lua:16>
	Last set from ~/.config/nvim/lua/config/options.lua line 14

```

**Reply by dpetka2001** [reply]
Just do `print("test") ` inside the callback.

**Reply by dpetka2001** [reply]
Try to do 
```lua
callback = function()
  vim.schedule(function()
    --your other logic here
  end)
end,
```
In case the `ObsidianToday` command is not available yet.

**Reply by dmyyy** [reply]
Doesn't work unfortunately - trying to print in the callback doesn't show me any messages anywhere. Running `:message` doesn't show me anything either.
